### ISSUE TITLE
Is there a service to get the available routes?

### ISSUE THREAD

**Issue by smasherprog** [OData/WebApi#1285]
Net core Odata  question:
Is there an available service for getting the routes for Odata similiar to IActionDescriptorCollectionProvider  that will provide the basics? I tried to find something like this, but I was unable to. 

### Assemblies affected

Odata beta 2 

### Expected result

A service that allows me to enumerate the available odata routes.

### Additional detail

The reason for this is that I need to build a list of available routes and assign permissions to those routes.

**Comment by robward-ms** [comment]
[PerRouteContainer.cs](https://github.com/OData/WebApi/blob/feature/netcore/src/Microsoft.AspNetCore.OData/PerRouteContainer.cs) has a mapping of all OData routes to their service containers but currently does not contain an enumeration capability. Is this the kind of information you are looking for?

**Comment by smasherprog** [comment]
Yeah. Just the meta data for each route. Post patch delete.  Controller action etc.    This is useful because I want to attach permissions dynamically to available routes so I need to get that information similar to how mvc exposes it.

**Comment by robward-ms** [comment]
We don't have a similar interface. Routing is done dynamically using the routing conventions and routes are re-discovered every time. OData could do better and maybe provide the feature you ask for so I'll mark this as a feature.

**Comment by smasherprog** [comment]
I didnt think that routes were discovered each time a url is requested since the information is available at startup. I'd like to offer a PR to build the routes just once at startup but I dont know the codebase well enough :(

**Comment by smasherprog** [comment]
thanks for adding it as a feature!
